core/src/main/scala/kafka/coordinator/transaction/TransactionLog.scala

```diff
@@ -120,21 +134,21 @@ object TransactionLogManager {
     *
     * @return value payload bytes
     */
-  private[coordinator] def valueToBytes(pidMetadata: PidMetadata): Array[Byte] = {
+  private[coordinator] def valueToBytes(txnMetadata: TransactionMetadata): Array[Byte] = {
     val value = new Struct(CURRENT_VALUE_SCHEMA)
-    value.set(VALUE_SCHEMA_PID_FIELD, pidMetadata.pid)
-    value.set(VALUE_SCHEMA_EPOCH_FIELD, pidMetadata.epoch)
-    value.set(VALUE_SCHEMA_TXN_TIMEOUT_FIELD, pidMetadata.txnTimeoutMs)
-    value.set(VALUE_SCHEMA_TXN_STATUS_FIELD, pidMetadata.txnMetadata.state.byte)
+    value.set(VALUE_SCHEMA_PID_FIELD, txnMetadata.pid)
+    value.set(VALUE_SCHEMA_EPOCH_FIELD, txnMetadata.epoch)
+    value.set(VALUE_SCHEMA_TXN_TIMEOUT_FIELD, txnMetadata.txnTimeoutMs)
+    value.set(VALUE_SCHEMA_TXN_STATUS_FIELD, txnMetadata.state.byte)
 
-    if (pidMetadata.txnMetadata.state.equals(NotExist)) {
-      if (pidMetadata.txnMetadata.topicPartitions.nonEmpty)
-        throw new IllegalStateException(s"Transaction is not expected to have any partitions since its state is ${pidMetadata.txnMetadata.state}: ${pidMetadata.txnMetadata}")
+    if (txnMetadata.state.equals(NotExist)) {
```

Comment:
nit: could we just do `txnMetadata == NotExist`? Also, maybe now it's safe to change `NotExist` to `Empty`?